### PR TITLE
[INV-86] Create review module, entity and migration

### PR DESCRIPTION
**Pull request by muhammadjon8**
image

### CODE REVIEW

File: database/migrations/1732127232754-create-review-entity.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+          },
+          {
+            name: 'rating',
+            type: 'int',
+            isNullable: false,
+          },
+          {
+            name: 'account_id',
+            type: 'uuid',
+            isNullable: true,
```
Comment (NegatiV-64) [inline database/migrations/1732127232754-create-review-entity.ts:46]: It shouldn't be nullable

File: database/migrations/1732127232754-create-review-entity.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+          },
+          {
+            name: 'accommodation_id',
+            type: 'uuid',
+            isNullable: true,
```
Comment (NegatiV-64) [inline database/migrations/1732127232754-create-review-entity.ts:51]: It shouldn't be nullable

File: database/migrations/1732127232754-create-review-entity.ts [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+          },
+          {
+            name: 'booking_id',
+            type: 'uuid',
+            isNullable: true,
```
Comment (NegatiV-64) [inline database/migrations/1732127232754-create-review-entity.ts:56]: It shouldn't be nullable

**Review by NegatiV-64 — CHANGES_REQUESTED** [review]

File: src/modules/review/entities/review.entity.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  accountId: string;
+
+  @ManyToOne(() => Accommodation, { onDelete: 'SET NULL' })
+  @JoinColumn({ name: 'accommodation_id' })
+  accommodation: Accommodation;
+
+  @Column({ name: 'accommodation_id' })
+  accommodationId: string;
+
+  @ManyToOne(() => Booking, { onDelete: 'SET NULL' })
```
Comment (NegatiV-64) [inline src/modules/review/entities/review.entity.ts:48]: If you want to reset to null, then you should actually fix the Typescript types for the entity

File: database/migrations/1732127232754-create-review-entity.ts [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+      }),
+    );
+
+    await queryRunner.createForeignKey(
+      'review',
+      new TableForeignKey({
+        columnNames: ['accommodation_id'],
+        referencedColumnNames: ['id'],
+        referencedTableName: 'accommodation',
+        onDelete: 'SET NULL',
```
Comment (farrooh) [inline database/migrations/1732127232754-create-review-entity.ts:78]: It's more logical to cascade delete reviews. Who needs a review that doesn't have accommodation?

**Review by farrooh — CHANGES_REQUESTED** [review]

**Review by NegatiV-64 — APPROVED** [review]

**Review by farrooh — APPROVED** [review]

**Pull request merged**